homeassistant/components/zha/config_flow.py

```diff
@@ -0,0 +1,55 @@
+"""Config flow for ZHA."""
+from collections import OrderedDict
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+
+from .const import (
+    DOMAIN, BAUD_RATES, CONF_BAUDRATE, CONF_DATABASE, CONF_RADIO_TYPE,
+    CONF_USB_PATH, RadioType
+)
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class ZhaFlowHandler(config_entries.ConfigFlow):
+    """Handle a config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    async def async_step_user(self, user_input=None):
+        """Handle a zha config flow start."""
+        if self._async_current_entries():
+            return self.async_abort(reason='single_instance_allowed')
+
+        errors = {}
+
+        if user_input is not None:
+            return self.async_create_entry(
```

Comment:
Need to think about how to do it. Currently in "zigpy" there's no "check config" method to probe whether the port is configured correct and is actually connected to a ZHA controller. Not without trying to do the actual `app.startup()`

I haven't dug too deep exploring the idea, but 
1. `await radio.connect()` could confirm whether the selected port is an actual serial port. This does not guarantee the port is connected to the actual ZHA controller or the baudrate is matching the actual baudrate of the controller. (AFAIK EZSP based controllers have default rate of 57600 and implementation of xbee radio would actually configure the radio to the baudrate from the config file if there's no response from the radio)

2. `await APPLICATION_CONTROLLER.startup(auto_form=True)` -- this would test both: serial port and confirm serial port is connected to ZHA controller and the path to DB file is "writable" file.